app/javascript/app/document-capture/components/file-input.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import FileImage from './file-image';
+import useInstanceId from '../hooks/use-instance-id';
+
+/**
+ * Returns true if the given file object is an image, or false otherwise.
+ *
+ * @param {File} file File object to test.
+ *
+ * @return {boolean} Whether given file is an image.
+ */
+export function isImageFile(file) {
+  return /^image\//.test(file.type);
+}
+
+function FileInput({ label, hint, accept, value, onChange }) {
+  const instanceId = useInstanceId();
+  const inputId = `file-input-${instanceId}`;
+  const hintId = `${inputId}-hint`;
+
+  return (
+    <>
+      {/*
+       * Disable reason: The Airbnb configuration of the `jsx-a11y` rule is strict in that it
+       * requires _both_ the `for` attribute and nesting, to maximize support for assistive
+       * technology. By the standard, only one or the other should be required. A form layout which
+       * includes a hint following a label cannot be nested within the label without misidentifying
+       * the hint as part of the label, which is the markup currently supported by USWDS.
+       *
+       * See: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/718
+       * See: https://github.com/airbnb/javascript/pull/2136
+       */}
+      {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+      <label htmlFor={inputId} className="usa-label">
+        {label}
+      </label>
+      {hint && (
+        <span className="usa-hint" id={hintId}>
+          {hint}
+        </span>
+      )}
+      <div className="usa-file-input">
+        <div className="usa-file-input__target">
+          {value && (
+            <div className="usa-file-input__preview-heading">
+              Selected file <span className="usa-file-input__choose">Change file</span>
```

Comment:
Are we still in throwaway code mode? Phrased another way: should we be translating this copy?

Reply:
> Are we still in throwaway code mode?

Sort of. This is a pretty direct adaptation of the USWDS file input, where the specific phrasing may change based on designs ([example from Slack](https://gsa-tts.slack.com/archives/CEUQ9FXNJ/p1595534805129800?thread_ts=1595427041.108700&cid=CEUQ9FXNJ)). And in some cases we may be blocked by the ability to interpolate elements into the string from translations (related to https://github.com/uswds/uswds/issues/3445#issuecomment-663215232), which is a bit bigger scope than I'd hoped to bite off here.

It's becoming increasingly clear to me that it'd be good for me to clarify or at least obligate myself to some follow-up here. Elsewhere, I tried to manage this by [unimplemented test cases](https://github.com/18F/identity-idp/pull/3978/files#diff-dc4face0a8c4558519155669c389bc87R91-R93). Might also be a good thing to track as a ticket, or "TODO" messages in the code (though when I tried TODO in another place, CodeClimate flagged it 😅 ).

Reply:
> Sort of. This is a pretty direct adaptation of the USWDS file input, where the specific phrasing may change based on designs ([example from Slack](https://gsa-tts.slack.com/archives/CEUQ9FXNJ/p1595534805129800?thread_ts=1595427041.108700&cid=CEUQ9FXNJ)). And in some cases we may be blocked by the ability to interpolate elements into the string from translations (related to [uswds/uswds#3445 (comment)](https://github.com/uswds/uswds/issues/3445#issuecomment-663215232)), which is a bit bigger scope than I'd hoped to bite off here.

I localized the texts in 47b70e1819f53e196208c3efe5f011bb46482293 . In retrospect, the fact that the labels may change by design may prove to be beneficial in how the string keys are named, so as to make them generic, and to constrain ourselves somewhat to adjust texts only to a reasonable extent where they still remain sensible under the generic key.

Element interpolation still remains an issue. For now, I've just used the plain text "Drag file or choose from folder" for the occurrence below. I have a solution in mind for how this might be implemented. I've opened LG-3237 as a follow-up task.